presto-hive/src/test/java/com/facebook/presto/hive/TestSymlinkTableListCaching.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import com.facebook.presto.hive.metastore.ExtendedHiveMetastore;
+import com.facebook.presto.hive.metastore.PrestoTableType;
+import com.facebook.presto.hive.metastore.PrincipalPrivileges;
+import com.facebook.presto.hive.metastore.Storage;
+import com.facebook.presto.hive.metastore.StorageFormat;
+import com.facebook.presto.hive.metastore.Table;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableMultimap;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.facebook.presto.hive.HiveQueryRunner.METASTORE_CONTEXT;
+import static com.facebook.presto.hive.HiveQueryRunner.getFileHiveMetastore;
+import static org.testng.Assert.assertEquals;
+
+public class TestSymlinkTableListCaching
+        extends AbstractTestQueryFramework
+{
+    ExtendedHiveMetastore metastore;
+    private static final PrincipalPrivileges PRINCIPAL_PRIVILEGES = new PrincipalPrivileges(ImmutableMultimap.of(), ImmutableMultimap.of());
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        ImmutableMap.Builder<String, String> extraHiveProperties = ImmutableMap.<String, String>builder()
+                .put("hive.file-status-cache-expire-time", "24h")
+                .put("hive.file-status-cache-size", "100000000")
+                .put("hive.file-status-cache-tables", "*");
+
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(
+                ImmutableList.of(),
+                ImmutableList.of(),
+                ImmutableMap.of(),
+                ImmutableMap.of(),
+                "sql-standard",
+                extraHiveProperties.build(),
+                Optional.of(1),
+                Optional.empty(),
+                Optional.empty(),
+                Optional.empty(),
+                true);
+
+        // setup file metastore for creating table
+        metastore = getFileHiveMetastore(queryRunner);
```

Comment:
When we create a metastore here, it will create a new instance since this method is static. Instead you can create this before the query runner and use below method instead.
` public static DistributedQueryRunner createQueryRunner(
            Iterable<TpchTable<?>> tpchTables,
            Iterable<String> tpcdsTableNames,
            Map<String, String> extraProperties,
            Map<String, String> extraCoordinatorProperties,
            String security,
            Map<String, String> extraHiveProperties,
            Optional<Integer> workerCount,
            Optional<Path> dataDirectory,
            Optional<BiFunction<Integer, URI, Process>> externalWorkerLauncher,
            Optional<ExtendedHiveMetastore> externalMetastore)`

Reply:
getFileHiveMetastore method in HiveQueryRunner takes queryRunner as input so shifting it before queryRunner object creation in the test class wouldn't help.

I have modified the getFileHiveMetastore method in HiveQueryRunner class and reused the same in my tests.